ethereum/contracts/bridge/L1WethBridge.sol

```diff
@@ -46,24 +47,41 @@ contract L1WethBridge is IL1Bridge, AllowListed, ReentrancyGuard {
     /// @dev The smart contract that manages the list with permission to call contract functions
     IAllowList public immutable allowList;
 
-    /// @dev zkSync smart contract that is used to operate with L2 via asynchronous L2 <-> L1 communication
-    IZkSync public immutable zkSync;
+    /// @dev bridgehub smart contract that is used to operate with L2 via asynchronous L2 <-> L1 communication
+    IBridgehub public immutable bridgehub;
 
     /// @dev The address of deployed L2 WETH bridge counterpart
-    address public l2Bridge;
+    address public __DEPRECATED_l2Bridge;
 
     /// @dev The address of the WETH on L2
     address public l2WethAddress;
 
     /// @dev A mapping L2 batch number => message number => flag
     /// @dev Used to indicate that zkSync L2 -> L1 WETH message was already processed
-    mapping(uint256 => mapping(uint256 => bool)) public isWithdrawalFinalized;
+    mapping(uint256 => mapping(uint256 => bool)) public __DEPRECATED_isWithdrawalFinalized;
```

Comment:
Old withdrawals need to be able to processed even after the transition to hyperchains. So, when user requests the finalization of the withdrawal from zkSync Era this mapping should be used.

Reply:
Yes I haven't added that feature yet